crates/storage/db/src/implementation/mdbx/tx.rs

```diff
@@ -76,13 +82,22 @@ impl<K: TransactionKind> Tx<K> {
 
     /// Gets a table database handle if it exists, otherwise creates it.
     pub fn get_dbi<T: Table>(&self) -> Result<DBI, DatabaseError> {
-        match self.db_handles.lock()[T::TABLE as usize] {
-            Some(handle) => Ok(handle),
-            ref mut handle @ None => {
-                let db =
-                    self.inner.open_db(Some(T::NAME)).map_err(|e| DatabaseError::Open(e.into()))?;
-                Ok(*handle.insert(db.dbi()))
+        // TODO: Use `OnceLock::get_or_try_init` once it's stable.
+        let slot = &self.db_handles[T::TABLE as usize];
+        match slot.get() {
+            Some(handle) => Ok(*handle),
+            None => self.open_and_store_db::<T>(slot),
+        }
+    }
+
+    #[cold]
+    fn open_and_store_db<T: Table>(&self, slot: &OnceLock<DBI>) -> Result<DBI, DatabaseError> {
+        match self.inner.open_db(Some(T::NAME)) {
+            Ok(db) => {
+                slot.set(db.dbi()).unwrap();
```

Comment:
can we map_err

Reply:
Already tried, this ends up as `map(|db| db.dbi()).map_err(..).inspect(slot.set)`, it's not any better